src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddlewareImpl.cs

```diff
@@ -192,7 +192,7 @@ private async Task HandleException(HttpContext context, ExceptionDispatchInfo ed
                     {
                         HttpContext = context,
                         AdditionalMetadata = exceptionHandlerFeature.Endpoint?.Metadata,
-                        ProblemDetails = { Status = DefaultStatusCode },
+                        ProblemDetails = { Status = context.Response.StatusCode },
```

Comment:
We should explicitly add a ProblemDetails test like https://github.com/dotnet/aspnetcore/blob/main/src/Middleware/Diagnostics/test/FunctionalTests/ProblemDetailsExceptionHandlerSampleTest.cs since that was one of the reasons for adding this API.

Reply:
Extended the sample to include a new endpoint `/throw/conflict` which responds with http status code 409. Tested this behaviour in a new functional test.